<html>
<head><meta charset="utf-8"><title>Provably optimal Needleman-Wunch · helpdesk (published) · Zulip Chat Archive</title></head>
<h2>Stream: <a href="https://juliacommunity.github.io/zulip-archive/stream/274208-helpdesk-.28published.29/index.html">helpdesk (published)</a></h2>
<h3>Topic: <a href="https://juliacommunity.github.io/zulip-archive/stream/274208-helpdesk-.28published.29/topic/Provably.20optimal.20Needleman-Wunch.html">Provably optimal Needleman-Wunch</a></h3>

<hr>

<base href="https://julialang.zulipchat.com">

<head><link href="https://juliacommunity.github.io/zulip-archive/style.css" rel="stylesheet"></head>

<a name="242979303"></a>
<h4><a href="https://julialang.zulipchat.com#narrow/stream/274208-helpdesk%20%28published%29/topic/Provably%20optimal%20Needleman-Wunch/near/242979303" class="zl"><img src="https://juliacommunity.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jakob Nybo Nissen <a href="https://juliacommunity.github.io/zulip-archive/stream/274208-helpdesk-.28published.29/topic/Provably.20optimal.20Needleman-Wunch.html#242979303">(Jun 17 2021 at 06:52)</a>:</h4>
<p><span class="user-mention" data-user-id="217836">@Kevin Bonham</span> interesting paper, that. I never heard about BURST before. Do you know of any papers that measure how much better BURST is than other (k-mer based) aligners in e.g. a metagenomic setting or at various levels of nucleotide identity?  The BURST paper itself skips _very_ lightly over the methodology.<br>
I mean, if you have a 250 bp read with 80% nt identity, there would be only 6.7e-5 probability of a length 12 kmer not matching if the mismatches were randomly distributed. And in bacteria, my experience is that there are usually some more conserved areas around the interesting parts of the genome anyway.</p>



<a name="242979406"></a>
<h4><a href="https://julialang.zulipchat.com#narrow/stream/274208-helpdesk%20%28published%29/topic/Provably%20optimal%20Needleman-Wunch/near/242979406" class="zl"><img src="https://juliacommunity.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jakob Nybo Nissen <a href="https://juliacommunity.github.io/zulip-archive/stream/274208-helpdesk-.28published.29/topic/Provably.20optimal.20Needleman-Wunch.html#242979406">(Jun 17 2021 at 06:54)</a>:</h4>
<p>(That being said, for some applications, if I could get even 10% of the reads to align more accurately, I would happily pay 10x the compute for that)</p>



<a name="243002042"></a>
<h4><a href="https://julialang.zulipchat.com#narrow/stream/274208-helpdesk%20%28published%29/topic/Provably%20optimal%20Needleman-Wunch/near/243002042" class="zl"><img src="https://juliacommunity.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Kevin Bonham <a href="https://juliacommunity.github.io/zulip-archive/stream/274208-helpdesk-.28published.29/topic/Provably.20optimal.20Needleman-Wunch.html#243002042">(Jun 17 2021 at 11:10)</a>:</h4>
<p><span class="user-mention" data-user-id="272650">@Jakob Nybo Nissen</span> I have not seen anyone use it in the wild. At a conference they presented some really impressive benchmarks - the context was shallow shotgun gut metagenomes, and the claim was that BURST allowed them to get remarkably good taxonomic assignment with like 20% of the number of reads.</p>
<p>Which would be huge, if true. But the last time I looked at the paper and the documentation, it was not obvious to me what steps to take to reproduce that.</p>



<a name="243002263"></a>
<h4><a href="https://julialang.zulipchat.com#narrow/stream/274208-helpdesk%20%28published%29/topic/Provably%20optimal%20Needleman-Wunch/near/243002263" class="zl"><img src="https://juliacommunity.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jakob Nybo Nissen <a href="https://juliacommunity.github.io/zulip-archive/stream/274208-helpdesk-.28published.29/topic/Provably.20optimal.20Needleman-Wunch.html#243002263">(Jun 17 2021 at 11:12)</a>:</h4>
<p>The reason I'm somewhat skeptical is that, if it was just a matter of more compute, people would have run Needleman-Wunch on reads already. I don't know how much effort and money it takes to obtain, say 200 gut metagenome NGS runs, but I'm betting it's a lot. Spending tens of thousands of dollars on that, and then not wanting to pay for a server to sit and run N-W for a few days would be weird.</p>



<a name="243002299"></a>
<h4><a href="https://julialang.zulipchat.com#narrow/stream/274208-helpdesk%20%28published%29/topic/Provably%20optimal%20Needleman-Wunch/near/243002299" class="zl"><img src="https://juliacommunity.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Kevin Bonham <a href="https://juliacommunity.github.io/zulip-archive/stream/274208-helpdesk-.28published.29/topic/Provably.20optimal.20Needleman-Wunch.html#243002299">(Jun 17 2021 at 11:12)</a>:</h4>
<p>I know Dan Knights (the PI) has a company (or consults with one maybe) that does analysis of microbiome metagenomes in a commercial setting, and I think they're using BURST for that, but I'm not sure</p>



<a name="243002448"></a>
<h4><a href="https://julialang.zulipchat.com#narrow/stream/274208-helpdesk%20%28published%29/topic/Provably%20optimal%20Needleman-Wunch/near/243002448" class="zl"><img src="https://juliacommunity.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Kevin Bonham <a href="https://juliacommunity.github.io/zulip-archive/stream/274208-helpdesk-.28published.29/topic/Provably.20optimal.20Needleman-Wunch.html#243002448">(Jun 17 2021 at 11:14)</a>:</h4>
<p>Well yeah, I mean that's what they were hyping - a revolution that gets you the precision of NW at drastically higher speeds. But it's tough to evaluate that claim of I can't run it <span aria-label="shrug" class="emoji emoji-1f937" role="img" title="shrug">:shrug:</span></p>



<hr><p>Last updated: Mar 02 2026 at 05:32 UTC</p>
</html>